server/filestore.go

```diff
@@ -474,6 +474,9 @@ func newFileStoreWithCreated(fcfg FileStoreConfig, cfg StreamConfig, created tim
 		// Check if our prior state remembers a last sequence past where we can see.
 		if fs.ld != nil && prior.LastSeq > fs.state.LastSeq {
 			fs.state.LastSeq, fs.state.LastTime = prior.LastSeq, prior.LastTime
+			if fs.state.Msgs == 0 {
+				fs.state.FirstSeq = fs.state.LastSeq + 1
```

Comment:
Should `fs.state.FirstTime` be set as well?
```go
fs.state.FirstTime = time.Time{}
```

Reply:
Good call.